src/util/spreadsheet/misc.py

```diff
@@ -0,0 +1,18 @@
+import json
+
+
+def iter_commands(cr, revision_ids):
+    if not revision_ids:
+        return
+    cr.execute("SELECT id, commands FROM spreadsheet_revision WHERE id IN %s", [tuple(revision_ids)])
+    for revision_id, data in cr.fetchall():
+        data_loaded = json.loads(data)
+        data_old = json.dumps(data_loaded)
+
+        if "commands" in data_loaded:
+            yield data_loaded["commands"]
+
+        data_new = json.dumps(data_loaded)
+        if data_new != data_old:
```

Comment:
Not totally true, the keys can be in another order.

Reply:
Added `sort_keys=True`. Still saving the original order to keep consistency with current scripts. It may also minimize the diff in the revision when seen from the users' perspective.